iet.so/src/action.cpp

```diff
@@ -413,8 +413,9 @@ void iet_action::hip_to_smi_indices(void) {
 
         // compute device location_id (needed to match this device
         // with one of those found while querying the pci bus
-        uint16_t hip_dev_location_id =
-            ((((uint16_t) (props.pciBusID)) << 8) | (((uint16_t)(props.pciDeviceID)) << 3) );
+        uint64_t hip_dev_location_id = ( ( ((uint64_t)props.pciDomainID & 0xffff ) << 32) |
```

Comment:
Why don't add (OR) function bits as well ? for its completeness.

Reply:
Left shifting **_props.pciDomainID_** by 16 bits should be enough. Right ?

Reply:
Since we are calling the id to index match based on smi APIs, I am using the same pattern as SMI uses to generate a id from Domain:Bus:Device enumerations. Also, strictly speaking we need to have another part for Function, which always if 0 in our case and hence we ignored it all this while and only 16 bit number was sufficient to encode B:D values. Now we are considering domain too , using 64 bit integer makes sense and using top 32 bits for domain is more sensible and gives bottom 32 bits for B:D:F values(9+ 6+ 4 = 19 bits which when aligned becomes 32 bits).

Reply:
I dont get the OR 3 function bits part? We are ORing 3 values alright here?

Reply:
> I dont get the OR 3 function bits part? We are ORing 3 values alright here?

I meant function bits in Bus:Device.Function format. Seems hip interface is not providing it. So, we can ignore it for now.

Reply:
> Since we are calling the id to index match based on smi APIs, I am using the same pattern as SMI uses to generate a id from Domain:Bus:Device enumerations. Also, strictly speaking we need to have another part for Function, which always if 0 in our case and hence we ignored it all this while and only 16 bit number was sufficient to encode B:D values. Now we are considering domain too , using 64 bit integer makes sense and using top 32 bits for domain is more sensible and gives bottom 32 bits for B:D:F values(9+ 6+ 4 = 19 bits which when aligned becomes 32 bits).

Ok. Let us follow the same format for PCIe id as smi.

Reply:
From rsmi_dev_pci_id_get() API documentation, it seems that for generating bdfid smi is using function bits as well. Need to check how are they able to get it as HPI API doesn't provide it.

Reply:
one can get these values from sys fs entry for that device's properties